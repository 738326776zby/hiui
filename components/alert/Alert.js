import React, { Component } from 'react'
import classNames from 'classnames'
import Icon from '../icon'
import './style/index'
class Alert extends Component {
  constructor (props) {
    super(props)
    this.state = { visible: true }
  }
  componentDidMount () {
    if (this.props.duration !== null) {
      window.setTimeout(() => {
        this.handleClose()
      }, this.props.duration)
    }
  }
  handleClose = () => {
    this.setState({ visible: false })
    this.props.onClose && this.props.onClose()
  }
  render () {
    const { prefixCls, type, theme, title, content, closeable } = this.props
    let classnames = classNames(prefixCls, `${prefixCls}--${type}`, `theme__${theme}`)
    let _type
    switch (type) {
      case 'warning':
        _type = 'exclamation-circle'
        break
      case 'error':
        _type = 'close-circle'
        break
      case 'success':
        _type = 'check-circle'
        break
      default:
        _type = 'info-circle'
    }

    return (
      this.state.visible && (
        <div className={classnames}>
          <div className='hi-alert__title'>
            <Icon name={_type} filled />
            {title}
          </div>
          {content && <div className='hi-alert__content'>{content}</div>}
          {closeable && <Icon name='close' className='hi-alert__close-btn' onClick={this.handleClose} />}
        </div>
      )
    )
  }
}

Alert.defaultProps = {
  prefixCls: 'hi-alert',
  type: 'info',
  closeable: true,
  duration: null
}

export default Alert
