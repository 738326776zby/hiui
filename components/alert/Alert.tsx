import * as React from 'react';
import classNames from 'classnames'
import './style/index'

class Alert extends React.Component<AlertProps,AlertState>  {
  state = { visible: true }
  static defaultProps = {
    prefixCls: 'hi-alert',
    type: 'info',
    closeable: true,
    duration: null
  }
  timeoutId :number
  componentDidMount () {
    if (this.props.duration !== null) {
    this.timeoutId = window.setTimeout(() => {
        this.handleClose()
      }, this.props.duration)
    }
  }
  componentWillUnmount () {
    clearTimeout(this.timeoutId)
  }
  handleClose () {
    this.setState({ visible: false })
    this.props.onClose && this.props.onClose()
  }
  render () {
    let classnames = classNames(this.props.prefixCls, this.state.visible, this.props.type, {
      noTitle: !this.props.title
    })
    let _type :string = 'tishi'

    switch (this.props.type) {
      case 'warning':
        _type = 'jinggao'
        break
      case 'error':
        _type = 'shibai'
        break
      case 'success':
        _type = 'chenggong'
        break
      default:
        _type = 'tishi'
    }

    return (
      this.state.visible && (
        <div className={classnames}>
          <div className='hi-icon__title'>
            <i className={`hi-icon icon-${_type}`} />
            {this.props.title && <div className='text-title'>{this.props.title}</div>}
          </div>
          {this.props.content && <div className='text-message'>{this.props.content}</div>}
          {this.props.closeable && (
            <div className='close-btn icon-img-delete' onClick={this.handleClose.bind(this)}>
              <i className='hi-icon icon-close' />
            </div>
          )}
        </div>
      )
    )
  }
}

export interface AlertProps {
    type ?: 'info' | 'error' |  'success' | 'warning';
    onClose ?: () => void;
    content ?: React.ReactNode;
    title ?: React.ReactNode;
    closeable ?: boolean;
    duration ?: number | null;
    prefixCls ?: 'hi-alert';
}
export interface AlertState {
    visible : boolean
}

export default Alert
