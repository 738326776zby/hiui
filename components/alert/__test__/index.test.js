import React from 'react'
import { mount } from 'enzyme'
import Alert from '../alert'

describe('Alert', () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it('could be closed', () => {
    const onClose = jest.fn()
    const wrapper = mount(<Alert type="info" message="信息提示的文案" onClose={onClose} />)
    wrapper.find('.close-btn').simulate('click')
    expect(onClose).toBeCalled()
    console.log(wrapper.find('.hi-alert'))
    expect(wrapper.find('.hi-alert')).toHaveLength(0)
  })


  it('auto close', () => {
    const onClose = jest.fn()
    const wrapper = mount(
      <Alert type="info" message="信息提示的文案" autoClose closeable={false} onClose={onClose} />
    )
    expect(wrapper.find('.hi-alert')).toHaveLength(1)
    expect(onClose).not.toBeCalled()
    jest.runAllTimers()
    expect(onClose).toBeCalled()
    console.dir(wrapper.find('.hi-alert'))

  })
})
