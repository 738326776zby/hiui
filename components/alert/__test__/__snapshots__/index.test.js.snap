// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert basic use 1`] = `
<div>
  <div
    className="hi-alert info middle noTitle"
  >
    <div
      className="icon-contain"
    >
      <i
        className="hi-icon icon-info-circle-o"
      />
    </div>
    <div
      className="text-contain"
    >
      <div
        className="text-message"
      >
        新提示
      </div>
    </div>
    <div
      className="close-btn icon-img-delete"
      onClick={[Function]}
    >
      <i
        className="hi-icon icon-close"
      />
    </div>
  </div>
  <div
    className="hi-alert success middle noTitle"
  >
    <div
      className="icon-contain"
    >
      <i
        className="hi-icon icon-check-circle-o"
      />
    </div>
    <div
      className="text-contain"
    >
      <div
        className="text-message"
      >
        成功
      </div>
    </div>
    <div
      className="close-btn icon-img-delete"
      onClick={[Function]}
    >
      <i
        className="hi-icon icon-close"
      />
    </div>
  </div>
  <div
    className="hi-alert error middle noTitle"
  >
    <div
      className="icon-contain"
    >
      <i
        className="hi-icon icon-close-circle-o"
      />
    </div>
    <div
      className="text-contain"
    >
      <div
        className="text-message"
      >
        错误
      </div>
    </div>
    <div
      className="close-btn icon-img-delete"
      onClick={[Function]}
    >
      <i
        className="hi-icon icon-close"
      />
    </div>
  </div>
  <div
    className="hi-alert warning middle noTitle"
  >
    <div
      className="icon-contain"
    >
      <i
        className="hi-icon icon-info-circle-o"
      />
    </div>
    <div
      className="text-contain"
    >
      <div
        className="text-message"
      >
        警告
      </div>
    </div>
    <div
      className="close-btn icon-img-delete"
      onClick={[Function]}
    >
      <i
        className="hi-icon icon-close"
      />
    </div>
  </div>
</div>
`;
